### ISSUE TITLE
Alexa can get events from the previous day

### ISSUE THREAD

**Issue by eBucher** [370-Alexa-Project/CS370_Echo_Demo#70]
Sometimes Alexa will get events from the a day prior to the given date(s) that the user wants to hear about. In the past we have had similar issues, and the solution was to add **::timestamp at time zone 'America/Los_Angeles'** to the end of the dates that we were using in our queries. Below is an example of an interaction that causes Alexa to get events from the wrong day.

--------------------------------

- **User:** What is happening on November fourteenth
(Note, the fourteenth is a Monday)

- **Alexa:** <speak>I was able to find 15 different events on Monday, <say-as interpret-as=\"date\">????1114</say-as>. What kind of events would you like to hear about?</speak>

- **User:** All of them

- **Alexa:** <speak>Cool. Here are the  events that I was able to find. On **Sunday**, <say-as interpret-as=\"date\">????1113</say-as> there is , Sonoma Film Institute at <say-as interpret-as=\"time\">4:00 PM</say-as>, Alpha Xi Delta Chapter Meeting Fall 2016 at <say-as interpret-as=\"time\">5:00 PM</say-as>, Fall 2016 Gamma Phi Beta Chapter Meeting at <say-as interpret-as=\"time\">6:00 PM</say-as>, Phi Delta Theta Weekly Chapter Meeting at <say-as interpret-as=\"time\">6:00 PM</say-as>, Alpha Psi Chapter Meeting Fall 2016 at <say-as interpret-as=\"time\">6:00 PM</say-as>, Alpha Gamma Delta Fall 2016 Chapter Meetings at <say-as interpret-as=\"time\">7:00 PM</say-as>, Intermural Basketball at <say-as interpret-as=\"time\">7:00 PM</say-as>, Phi Sigma Sigma Chapter Meeting at <say-as interpret-as=\"time\">7:00 PM</say-as>, Pi Kappa Phi Chapter Meeting Fall 2016 at <say-as interpret-as=\"time\">7:04 PM</say-as>, Intermural Indoor Soccer at <say-as interpret-as=\"time\">8:00 PM</say-as>, Alpha Psi New Member Meeting Fall 2016 at <say-as interpret-as=\"time\">8:00 PM</say-as>, and Kappa Delta Zeta New Member Meeting at <say-as interpret-as=\"time\">8:30 PM</say-as>. On **Monday**, <say-as interpret-as=\"date\">????1114</say-as> there is Fall 2016 Monthly Nursing Club Meeting at <say-as interpret-as=\"time\">10:00 AM</say-as>, Bible Club at SSU Bible Study at <say-as interpret-as=\"time\">11:00 AM</say-as>, and Women's Ultimate Frisbee Club Meetings at <say-as interpret-as=\"time\">12:00 PM</say-as>.</speak>

------------------------------------------------------------------------------

**Comment by eBucher** [comment]
Lambda Request JSON after the user says "All of them":

```
{
  "session": {
    "sessionId": "SessionId.af1437ba-f1be-42c6-acde-b9984d7c4171",
    "application": {
      "applicationId": "amzn1.ask.skill.68a28ff8-30a5-493e-8e11-fca2cb138d6c"
    },
    "attributes": {
      "recentlySaidEvents": {
        "Fall 2016 Monthly Nursing Club Meeting": 209,
        "Alpha Xi Delta Chapter Meeting Fall 2016": 199,
        "Alpha Psi Chapter Meeting Fall 2016": 200,
        "Phi Delta Theta Weekly Chapter Meeting": 202,
        "IMS Basketball": 99,
        "Phi Sigma Sigma Chapter Meeting": 205,
        "Sonoma Film Institute": 26,
        "Alpha Psi New Member Meeting Fall 2016": 207,
        "Pi Kappa Phi Chapter Meeting Fall 2016": 206,
        "Women's Ultimate Frisbee Club Meetings": 211,
        "AGD Fall 2016 Chapter Meetings": 203,
        "Bible Club at SSU Bible Study": 210,
        "KDZ New Member Meeting": 208,
        "IMS Indoor Soccer": 100,
        "Fall 2016 Gamma Phi Beta Chapter Meeting": 201
      },
      "stateId": "LIST_TOO_LONG",
      "savedDate": {
        "relativeDate": "Monday, <say-as interpret-as=\"date\">????1114</say-as>",
        "relativeDateWithPreposition": "on Monday, <say-as interpret-as=\"date\">????1114</say-as>",
        "begin": "2016-11-14",
        "end": "2016-11-15"
      }
    },
    "user": {
      "userId": "amzn1.ask.account.AGUBCBZQXDWUBLXZHDQPT2U77RUUKFW7JWE5TRFZ6UH62H47ZP43CRKNOLYE27GK6BF2FUOJF3UTBBMI4ET4WEIWVPMPH72MGCLKVBORDBGU6O6PFXCD33S2R6GKKXI426Y4XI4RGJ4ULN3J2I3EVHNPJK2ELLUTGLLUFY26C3U6FDUKLID5ZP7EX4TS6JTKKWQKJUDBSI4DRAQ"
    },
    "new": false
  },
  "request": {
    "type": "IntentRequest",
    "requestId": "EdwRequestId.6cfa0d7f-3bfe-4120-b10f-dce5cec3c107",
    "locale": "en-US",
    "timestamp": "2016-11-11T03:08:31Z",
    "intent": {
      "name": "AllCategoryIntent",
      "slots": {}
    }
  },
  "version": "1.0"
}
```

Lambda Response

```
{
  "version": "1.0",
  "response": {
    "outputSpeech": {
      "type": "SSML",
      "ssml": "<speak>Cool. Here are the  events that I was able to find. On Sunday, <say-as interpret-as=\"date\">????1113</say-as> there is , Sonoma Film Institute at <say-as interpret-as=\"time\">4:00 PM</say-as>, Alpha Xi Delta Chapter Meeting Fall 2016 at <say-as interpret-as=\"time\">5:00 PM</say-as>, Fall 2016 Gamma Phi Beta Chapter Meeting at <say-as interpret-as=\"time\">6:00 PM</say-as>, Phi Delta Theta Weekly Chapter Meeting at <say-as interpret-as=\"time\">6:00 PM</say-as>, Alpha Psi Chapter Meeting Fall 2016 at <say-as interpret-as=\"time\">6:00 PM</say-as>, Alpha Gamma Delta Fall 2016 Chapter Meetings at <say-as interpret-as=\"time\">7:00 PM</say-as>, Intermural Basketball at <say-as interpret-as=\"time\">7:00 PM</say-as>, Phi Sigma Sigma Chapter Meeting at <say-as interpret-as=\"time\">7:00 PM</say-as>, Pi Kappa Phi Chapter Meeting Fall 2016 at <say-as interpret-as=\"time\">7:04 PM</say-as>, Intermural Indoor Soccer at <say-as interpret-as=\"time\">8:00 PM</say-as>, Alpha Psi New Member Meeting Fall 2016 at <say-as interpret-as=\"time\">8:00 PM</say-as>, and Kappa Delta Zeta New Member Meeting at <say-as interpret-as=\"time\">8:30 PM</say-as>. On Monday, <say-as interpret-as=\"date\">????1114</say-as> there is Fall 2016 Monthly Nursing Club Meeting at <say-as interpret-as=\"time\">10:00 AM</say-as>, Bible Club at SSU Bible Study at <say-as interpret-as=\"time\">11:00 AM</say-as>, and Women's Ultimate Frisbee Club Meetings at <say-as interpret-as=\"time\">12:00 PM</say-as>.</speak>"
    },
    "reprompt": {
      "outputSpeech": {
        "type": "SSML",
        "ssml": "<speak>Is there anything you would like to know about those events?</speak>"
      }
    },
    "shouldEndSession": false
  },
  "sessionAttributes": {
    "recentlySaidEvents": {
      "Fall 2016 Monthly Nursing Club Meeting": 209,
      "Alpha Xi Delta Chapter Meeting Fall 2016": 199,
      "Alpha Psi Chapter Meeting Fall 2016": 200,
      "Phi Delta Theta Weekly Chapter Meeting": 202,
      "IMS Basketball": 99,
      "Phi Sigma Sigma Chapter Meeting": 205,
      "Sonoma Film Institute": 26,
      "Alpha Psi New Member Meeting Fall 2016": 207,
      "Pi Kappa Phi Chapter Meeting Fall 2016": 206,
      "Women's Ultimate Frisbee Club Meetings": 211,
      "AGD Fall 2016 Chapter Meetings": 203,
      "Bible Club at SSU Bible Study": 210,
      "KDZ New Member Meeting": 208,
      "IMS Indoor Soccer": 100,
      "Fall 2016 Gamma Phi Beta Chapter Meeting": 201
    },
    "stateId": "USER_HEARD_EVENTS",
    "savedDate": {
      "relativeDate": "Monday, <say-as interpret-as=\"date\">????1114</say-as>",
      "relativeDateWithPreposition": "on Monday, <say-as interpret-as=\"date\">????1114</say-as>",
      "begin": "2016-11-14",
      "end": "2016-11-15"
    }
  }
}
```

**Comment by ghost** [comment]
This should be fixed by 1cdcf8112c96342c7d00a489ecf289099efeb88f
